src/libraries/Directory.Build.targets

```
@@ -33,8 +33,13 @@
          SYSLIB0036: Regex.CompileToAssembly
     -->
     <NoWarn Condition="'$(IsPartialFacadeAssembly)' == 'true'">$(NoWarn);SYSLIB0003;SYSLIB0004;SYSLIB0015;SYSLIB0017;SYSLIB0021;SYSLIB0022;SYSLIB0023;SYSLIB0025;SYSLIB0032;SYSLIB0036</NoWarn>
-    <!-- Reset these properties back to blank, since they are defaulted by Microsoft.NET.Sdk -->
-    <WarningsAsErrors Condition="'$(WarningsAsErrors)' == 'NU1605'" />
+    <!-- Microsoft.NET.Sdk enables some warnings as errors out of the box.
+         We want to remove some items from this list so they don't fail the build.
+         Can't use 'WarningsNotAsErrors' element because vbproj doesn't honor it.
+         Items to remove:
+         NU1605: Package downgrade detected
+    -->
+    <WarningsAsErrors>$([System.Text.RegularExpressions.Regex]::Replace($(WarningsAsErrors), '\bNU1605\b', ''))</WarningsAsErrors>
```

I would prefer to not use a regex here if not strictly necessary. This is a code path that runs in every msbuild evaluation, a normal Replace should be sufficient. This will work as long as we have 4 digit nuget error codes. I would be surprised if we ever get to 5 digits.

```suggestion
    <WarningsAsErrors>$(WarningsAsErrors.Replace('NU1605', ''))</WarningsAsErrors>
```